doghouse/client/client.go

```diff
@@ -0,0 +1,77 @@
+package client
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+
+	"github.com/haya14busa/reviewdog"
+	"github.com/haya14busa/reviewdog/doghouse"
+)
+
+const baseEndpoint = "https://review-dog.appspot.com"
+
+type DogHouseClient struct {
+	Client *http.Client
+	// Base URL for API requests. Defaults is https://review-dog.appspot.com.
+	BaseURL *url.URL
+}
+
+func New(client *http.Client) *DogHouseClient {
+	dh := &DogHouseClient{Client: client}
+	if dh.Client == nil {
+		dh.Client = http.DefaultClient
+	}
+	base := baseEndpoint
+	if baseEnvURL := os.Getenv("REVIEWDOG_GITHUB_APP_URL"); baseEnvURL != "" {
+		base = baseEnvURL
+	}
+	var err error
+	dh.BaseURL, err = url.Parse(base)
+	if err != nil {
+		log.Fatal(err)
+	}
+	return dh
+}
+
+func (c *DogHouseClient) Check(ctx context.Context, req *doghouse.CheckRequest) (*doghouse.CheckResponse, error) {
```

Comment:
**[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported method DogHouseClient.Check should have comment or be unexported